src/e3/cve.py

```diff
@@ -79,6 +80,9 @@ def search_by_cpe_name(
         :param results_per_page: number of results to return for each request,
              note that it is recommended to keep the default setting
         """
+        if self.session is None:
+            raise ValueError("NVD session not started")
```

Comment:
What about: *NVD session not Stated: the NVD class should be used in a `with` statement* (or something like that)

Reply:
Done.